checkov/common/checks_infra/solvers/attribute_solvers/base_attribute_solver.py

```diff
@@ -199,11 +199,27 @@ def _is_variable_dependant(value: Any, source: str) -> bool:
         return False
 
     @staticmethod
-    def _render_json_str(value_to_check: Any, attr: str, vertex: Dict[str, Any]) -> Any:
-        if attr == 'policy' and vertex.get('resource_type', '').endswith('policy'):
-            try:
-                value_to_check = json.loads(value_to_check)
-                return value_to_check
-            except Exception as e:
-                logging.info(f'cant parse policy str to object, {str(e)}')
+    def _render_json_str(value_to_check: Any) -> Any:
+        """
+        Tries to render objects containing json dict, for example:
+        - resource "aws_iam_policy" "example1" {
+            policy == <<POLICY
+                {
+                ... json-data
+                }
+            POLICY
+        }
+        - resource "aws_emr_security_configuration" "example2" {
+            configuration = <<EOF
+                {
+                ... json-data
+                }
+            EOF
+        }
+        """
+        try:
+            value_to_check = json.loads(value_to_check)
```

Comment:
I'm worrying about performance degradation here, since we will try parsing every attribute into a JSON.
Implement a mechanism which lookups a JSON object starting delimiters (`{` for object , `[` for an array of objects), and parse into a JSON only when there's a match

Reply:
Do you think it will be better to do it like that if you also relate to what @gruebel said about this part?
I mainly try to understand if it is even ok to try to manipulate each attribute like that, even if it does look like a json object.

Reply:
if we target a more general solution then this would be the minimum to do to make sure, we don't try to parse everything as json.
the other approach would be to maintain a map of resource + argument, which should be parsed as json.